File: bindings/nodejs/examples/how_tos/wallet/send-amount.ts
// Copyright 2023 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import { Wallet, initLogger, Utils } from '@iota/sdk';

// This example uses secrets in environment variables for simplicity which should not be done in production.
//
// Make sure that `example.stronghold` and `example.walletdb` already exist by
// running the `how_tos/accounts_and_addresses/create-wallet` example!
//
require('dotenv').config({ path: '.env' });

// Run with command:
// yarn run-example ./how_tos/account_wallet/transaction.ts

// In this example we send funds from an account wallet.
async function run() {
    initLogger();
    try {
        for (const envVar of ['WALLET_DB_PATH', 'STRONGHOLD_PASSWORD']) {
            if (!(envVar in process.env)) {
                throw new Error(
                    `.env ${envVar} is undefined, see .env.example`,
                );
            }
        }

        const syncOptions = {
            account: {
                basicOutputs: true,
            },
        };

        const wallet = await Wallet.create({
            storagePath: process.env.WALLET_DB_PATH,
        });

        await wallet.setStrongholdPassword(
            process.env.STRONGHOLD_PASSWORD as string,
        );

        const balance = await wallet.sync(syncOptions);

        const totalBaseTokenBalance = balance.baseCoin.total;
        console.log(
            `Balance before sending funds from account: ${totalBaseTokenBalance}`,
        );

        const accountId = balance.accounts[0];
        console.log(`Account Id: ${accountId}`);

        const client = await wallet.getClient();

        // Get Account address
        const accountAddress = Utils.accountIdToBech32(
            accountId,
            await client.getBech32Hrp(),
        );

        // Find first output unlockable by the account address
        const queryParameters = {
            address: accountAddress,
        };
        const input = (await client.basicOutputIds(queryParameters)).items[0];

        const params = [
            {
                address:
                    'rms1qpszqzadsym6wpppd6z037dvlejmjuke7s24hm95s9fg9vpua7vluaw60xu',
                amount: BigInt(1000000),
            },
        ];
        const options = {
            mandatoryInputs: [input],
            allowMicroAmount: false,
        };
        const transaction = await wallet.sendWithParams(params, options);
        await wallet.reissueTransactionUntilIncluded(transaction.transactionId);
        console.log(
            `Transaction with custom input: https://explorer.iota.org/testnet/transaction/${transaction.transactionId}`,
        );

        const totalBaseTokenBalanceAfter = (await wallet.sync(syncOptions))
            .baseCoin.total;
        console.log(
            `Balance after sending funds from account: ${totalBaseTokenBalanceAfter}`,
        );
    } catch (error) {
        console.error('Error: ', error);
    }
}

void run().then(() => process.exit());

File: bindings/python/examples/how_tos/account_output/send_amount.py
import os

from dotenv import load_dotenv
from iota_sdk import AccountAddress, Wallet, WalletOptions, Utils, NodeIndexerAPI, SyncOptions, WalletSyncOptions, SendParams

# In this example we send funds from an account output.

load_dotenv()

for env_var in ['WALLET_DB_PATH', 'STRONGHOLD_PASSWORD']:
    if env_var not in os.environ:
        raise Exception(f".env {env_var} is undefined, see .env.example")

wallet = Wallet(WalletOptions(storage_path=os.environ.get('WALLET_DB_PATH')))

wallet.set_stronghold_password(os.environ["STRONGHOLD_PASSWORD"])

sync_options = SyncOptions(wallet=WalletSyncOptions(basic_outputs=True))
balance = wallet.sync(sync_options)

total_base_token_balance = balance.base_coin.total
print(
    f'Balance before sending funds from the account output: {total_base_token_balance}')

account_id = balance.accounts[0]
print(f'Account Id: {account_id}')

# Get account address
account_address = Utils.address_to_bech32(
    AccountAddress(account_id), wallet.get_client().get_bech32_hrp())

# Find first output unlockable by the account address
query_parameters = NodeIndexerAPI.BasicOutputQueryParameters(
    address=account_address)
inputs = [wallet.get_client().basic_output_ids(query_parameters).items[0]]

params = [SendParams(
    address='rms1qpszqzadsym6wpppd6z037dvlejmjuke7s24hm95s9fg9vpua7vluaw60xu',
    amount=1000000,
)]
options = {
    'requiredInputs': inputs,
}
transaction = wallet.send_with_params(params, options)
wallet.wait_for_transaction_acceptance(
    transaction.transaction_id)
print(
    f'Transaction with custom input: https://explorer.shimmer.network/testnet/transaction/{transaction.transaction_id}')

total_base_token_balance = wallet.sync(sync_options).base_coin.total
print(
    f'Balance after sending funds from the account output: {total_base_token_balance}')
